### PR TITLE
Moved GitHub repo activate/deactivate control to frontend

### LINKED ISSUE #137: Rebase sync github accounts button on Ajax
# Overview

A button `/github/settings -> Sync account` should be implemented using Ajax.

# Tasks

- [x] github api endpoints
- [x] sync account
- [x] activate
- [x] deactivate

### PR DESCRIPTION
**Pull request by roll**
- fixes #137

### CODE REVIEW

**Comment by roll** [comment]
@amercader 
please take a look

File: goodtablesio/integrations/github/blueprint.py [hunk 1]
```diff
@@ -265,3 +222,43 @@ def api_repo(repo_id):
...
     }), code)
+
+
+@github.route('/api/repo/<repo_id>/activate')
+@login_required
+def api_repo_activate(repo_id):
+    error = None
+    token = current_user.github_oauth_token
+    if not token:
+        error = 'No valid GitHub token found'
```
Comment (amercader) [inline goodtablesio/integrations/github/blueprint.py]: No need to continue the execution if there is no token right? just return the error

File: goodtablesio/integrations/github/blueprint.py [hunk 2]
```diff
@@ -265,3 +222,43 @@ def api_repo(repo_id):
...
+    repo = database['session'].query(GithubRepo).get(repo_id)
```
Comment (amercader) [inline goodtablesio/integrations/github/blueprint.py]: Handle repo not found

File: goodtablesio/integrations/github/blueprint.py [hunk 3]
```diff
@@ -265,3 +222,43 @@ def api_repo(repo_id):
...
+
+
+@github.route('/api/repo/<repo_id>/deactivate')
+@login_required
+def api_repo_deactivate(repo_id):
+    error = None
+    token = current_user.github_oauth_token
+    if not token:
+        error = 'No valid GitHub token found'
+    repo = database['session'].query(GithubRepo).get(repo_id)
```
Comment (amercader) [inline goodtablesio/integrations/github/blueprint.py]: Same as above (I know I probably wrote the initial implementation :) )

Comment (roll) [inline goodtablesio/integrations/github/blueprint.py]: I've badly ported it. For S3 I've control flow to check error but missed it here

**Comment by roll** [comment]
@amercader 
- added frontend api request error handlers
- restricted access to repos (everything not belonging to the user -> Forbidden 403)
- slightly updated API semantics based on your ideas - return json only when it's 200 (so we have fixable user error)

**Pull request merged**